docs/spatialexample.rst

```diff
@@ -368,3 +372,333 @@ Run Everything
+        large-scale spatial epidemic simulations.
+
+        Attributes:
+        ------------
+        model : object
+            The simulation model containing population and node information.
+        network : ndarray
+            A matrix representing the transmission probabilities between nodes.
+        locations : ndarray
+            Array of node latitude and longitude coordinates.
+        """
+        def __init__(self, model):
+            """
+            Initializes the transmission component by computing the infection migration network.
+
+            Parameters:
+            -----------
+            model : object
+                The simulation model containing population and node information.
+            """
+            self.model = model
+            model.nodes.add_vector_property("network", length=model.nodes.count, dtype=np.float32)
+            self.network = model.nodes.network
+
+            # Extract node locations and populations from model.population_data
+            self.locations = np.column_stack((model.population_data["centroid_lat"], model.population_data["centroid_long"]))
+            initial_populations = np.array(model.population_data["population"])
+
+            # Initialize heterogeneous transmission factor per agent (0.5 to 2.0)
+            self.model.population.tx_hetero_factor = np.random.uniform(0.5, 2.0, size=model.population.capacity)
+
+            # Compute the infection migration network based on node populations.
+            a, b, c, k = self.model.params.a, self.model.params.b, self.model.params.c, self.model.params.k
+
+            # Compute all pairwise distances in one call (this speeds up initialization significantly)
+            distances = calc_distances(self.locations[:, 0], self.locations[:, 1])
+            self.network = gravity(initial_populations, distances, k, a, b, c)
+            self.network /= np.power(initial_populations.sum(), c)  # Normalize
+            self.cap_network_flow()
+
+        def cap_network_flow(self):
+            """
+            Caps the total fraction of population that can migrate infection in a given timestep.
+            This prevents unrealistically high transmission rates in a single step.
+            """
+            max_frac = 0.01  # Limit transmission migration to prevent overflow
+            for row in range(self.network.shape[0]):
+                total_outflow = self.network[row].sum()
+                if total_outflow > max_frac:
+                    self.network[row] *= max_frac / total_outflow  # Normalize outgoing
+
+            # Also cap incoming infection migration to maintain balance
+            network_incoming = self.network.T
+            for col in range(network_incoming.shape[0]):
+                total_inflow = network_incoming[col].sum()
+                if total_inflow > max_frac:
+                    network_incoming[col] *= max_frac / total_inflow
```

Comment:
We already have a function for capping network flow: `row_normalizer()`.

https://github.com/InstituteforDiseaseModeling/laser/blob/main/src/laser_core/migration.py#L122